db/change_listener.go

```diff
@@ -212,10 +221,49 @@ func (listener *changeListener) Notify(ctx context.Context, keys channels.Set) {
 	}
 	base.DebugfCtx(ctx, base.KeyChanges, "Notifying that %q changed (keys=%q) count=%d",
 		base.MD(listener.bucketName), base.UD(keys), listener.counter)
-	listener.tapNotifier.Broadcast()
 	listener.tapNotifier.L.Unlock()
 }
 
+func (listener *changeListener) BroadcastChanges(ctx context.Context) {
+	ticker := time.NewTicker(DefaultBroadcastChangesTime)
+	// boolean to indicate whether ticker is using the default value, this is needed so we don't call reset on ticker
+	// for a value it already has
+	defaultTickerValue := true
+	terminator := listener.terminator // take copy of this to avoid race when restarting change listener in tests
```

Comment:
I don't think this makes a copy, this is a channel which is implemented as a pointer

Reply:
Might be clearer if this is passed into the goroutine function instead to capture it - it's functionally equivalent to copying the pointer address.

```go
go func(terminator chan bool) {
	// ...
}(listener.terminator)
```